### ISSUE TITLE
[GNS3 3.0.1] Cannot create a node template if the image is already on the controller

### ISSUE THREAD

**Issue by nwhisper2014** [GNS3/gns3-gui#3684]
**Describe the bug**
When trying to create a node template, the error message “The template cannot be created: Image ‘<some_image>’ could not be found in the controller database” appears if the image is already on the controller in the corresponding directory.

The images are loaded correctly and have correct access rights:
```
-rwx------ 1 gns3 gns3  57319424 Oct 24 09:48 vios-adventerprisek9-m.spa.159-3.m8.qcow2
-rw-r--r-- 1 gns3 gns3        32 Dec 28 11:19 vios-adventerprisek9-m.spa.159-3.m8.qcow2.md5sum

-rwx------ 1 gns3 gns3 240355720 Jun 19  2024 x86_64_crb_linux_l2-adventerprisek9-ms
-rw-r--r-- 1 gns3 gns3        32 Dec 28 11:28 x86_64_crb_linux_l2-adventerprisek9-ms.md5sum
```

**GNS3 version and operating system (please complete the following information):**
 - OS: Arch Linux (the gns3-gui client is connected to the controller via HTTPS encrypted connection)
 - GNS3 version: 3.0.1
 - Remote GNS3 VM on VirtualBox, KVM

**To Reproduce**
1 - upload the node image to the controller
2 - open the New template dialog
3 - select the template corresponding to the loaded image
4 - go through all the steps of the template creation wizard and press Finish button

**Screenshots or videos**
![image-not-found-on-controller1_0](https://github.com/user-attachments/assets/af727a2a-b6c6-4928-8bbf-4b143893c79c)
![image-not-found-on-controller1_1](https://github.com/user-attachments/assets/e84dba76-c425-4362-ad43-08c0881783a4)
![image-not-found-on-controller2_0](https://github.com/user-attachments/assets/5936f05d-6329-42c6-a65c-1b8d049f9685)
![image-not-found-on-controller2_1](https://github.com/user-attachments/assets/bd12ae11-6073-4ae5-a117-c56febfe7311)

**Comment by grossmj** [comment]
Please can you post a screenshot of the image manager as well? (In the GUI: File -> Image management)

![Image](https://github.com/user-attachments/assets/83defbd3-4633-4c9c-9cdd-d4aadeb55287)

Thanks.

**Comment by nwhisper2014** [comment]
> Please can you post a screenshot of the image manager as well? (In the GUI: File -> Image management)

![image-manager](https://github.com/user-attachments/assets/3b014bdd-88ff-42b6-b395-ab6bac95e6a2)

**Comment by ajitprof** [comment]
I ran into the same issue after I upgraded to 3.0.1. To resolve this you would need to upload the image to image management directory. Then when you import the template, the device should load it this time.

**Comment by nwhisper2014** [comment]
> I ran into the same issue after I upgraded to 3.0.1. To resolve this you would need to upload the image to image management directory. Then when you import the template, the device should load it this time.

I think the add templates dialog should not report the presence of the image on the controller if the old option of uploading the image to gns3vm via ssh is no longer a working option, as this is misleading and seems like a bug.

**Comment by nwhisper2014** [comment]
> Please can you post a screenshot of the image manager as well? (In the GUI: File -> Image management)

I forgot to add: the images that are not in the list are uploaded via ssh from the old gns3 server. It was as convenient and fast as possible to transfer via ssh.

If in the New еemplate dialog box I click Import and upload the image this way, no error appears and the template is successfully created.

**Comment by ajitprof** [comment]
> > I ran into the same issue after I upgraded to 3.0.1. To resolve this you would need to upload the image to image management directory. Then when you import the template, the device should load it this time.
> 
> 
> 
> I think the add templates dialog should not report the presence of the image on the controller if the old option of uploading the image to gns3vm via ssh is no longer a working option, as this is misleading and seems like a bug.

I agree. I would prefer the old way or an easier way to import the images when the template was launched. This helped me get back up and running because I was unable to import any devices before on 3.0.0. Thanks for reporting this.

**Comment by grossmj** [comment]
> I forgot to add: the images that are not in the list are uploaded via ssh from the old gns3 server. It was as convenient and fast as possible to transfer via ssh.

This should have worked however I found a problem that the controller was trying to add an image to the database before it was finished to be copied/uploaded. I fixed that in https://github.com/GNS3/gns3-server/pull/2471

> These images are not present in Image Manager, but the template wizard sees them and displays them as present on the controller. Hmm...

Indeed, there was a problem in the GUI that called the old API endpoints that listed images directly present on a compute and not the controller database. Thanks for reporting this! 👍

**Comment by Andre4s** [comment]
> > Забыл добавить: изображения, которых нет в списке, загружены по ssh со старого сервера gns3. Максимально удобно и быстро было передавать по ssh.
> 
> Это должно было сработать, однако я обнаружил проблему, что контроллер пытался добавить изображение в базу данных до того, как оно было скопировано/загружено. Я исправил это в [GNS3/gns3-server#2471](https://github.com/GNS3/gns3-server/pull/2471)
> 
> > Эти изображения отсутствуют в диспетчере изображений, но мастер шаблонов видит их и отображает как присутствующие на контроллере. Хм...
> 
> Действительно, была проблема в GUI, которая вызывала старые конечные точки API, которые перечисляли изображения, присутствующие непосредственно на вычислительном устройстве, а не в базе данных контроллера. Спасибо, что сообщили об этом! 👍

I still have this problem. on 3.0.3

**Comment by grossmj** [comment]
> I still have this problem. on 3.0.3

The fix made it in v3.0.4, please try with this version.

**Comment by peracchi** [comment]
Having the same problem in v3.0.5

**Comment by ivanobom** [comment]
> Having the same problem in v3.0.5

Mee, too. It is much more convenient to copy images to controller server via ssh and then just click on a "refresh" button for controller's database to be updated

**Comment by epiecs** [comment]
Hi,

Same problem here. I had a full ansible setup to deploy gns3 servers + upload all needed images for all the classes that i teach. Currently I am migrating everything over to api v3 (which up to this point is a breeze thanks to the nice docs). 

But I have the same issue. I have a central file server that I use to copy over the images to each gns3 instance but when I call the templates api I get the error that the image does not exist.

Uploading the image via the api would be cumbersome. Is there an option planned in the future where we can just do an api call to refresh all images in all folders?

**Comment by ItsAbderrahimEl** [comment]
gns3 version 3.0.6 same problem, I'm uploading manually a .iso file of VyOS and the same bug shows up: Image '/root/GNS3/images/QEMU/vyos-2026.09.09-0029-rolling-generic-amd64.iso' could not be found in the controller database
